api/src/main/java/jakarta/nosql/Template.java

```diff
@@ -0,0 +1,279 @@
+     * that changed due to the insert. After invoking this method, do not continue to use
+     * the entity instances that are supplied in the parameter. This method makes no guarantees
+     * about the state of the entity instances that are supplied in the parameter.
+     * The position of entities within the {@code Iterable} return value must correspond to the
+     * position of entities in the parameter based on the unique identifier of the entity.</p>
+     *
+     * @param entities entities to insert.
+     * @param <T>      the entity type
+     * @return an iterable containing the inserted entities, which may or may not be different instances depending
+     * on whether the insert caused values to be generated or automatically incremented.
+     * @throws NullPointerException if the iterable is null or any element is null.
+     */
+    <T> Iterable<T> insert(Iterable<T> entities);
+
+    /**
+     * Inserts multiple entities into the database with the expiration date. If any entity of this type with the same
+     * unique identifier as any of the given entities already exists in the database and the database
+     * supports ACID transactions, then this method raises an error.
+     * In databases that follow the BASE model or use an append model to write data, this exception
+     * is not thrown.
+     *
+     * <p>The entities within the returned {@link Iterable} must include all values that were
+     * written to the database, including all automatically generated values and incremented values
+     * that changed due to the insert. After invoking this method, do not continue to use
+     * the entity instances that are supplied in the parameter. This method makes no guarantees
+     * about the state of the entity instances that are supplied in the parameter.
+     * The position of entities within the {@code Iterable} return value must correspond to the
+     * position of entities in the parameter based on the unique identifier of the entity.</p>
+     *
+     * @param entities entities to insert.
+     * @param <T>      the entity type
+     * @param ttl      time to live
+     * @return an iterable containing the inserted entities, which may or may not be different instances depending
+     * on whether the insert caused values to be generated or automatically incremented.
+     * @throws NullPointerException if the iterable is null or any element is null.
+     */
+    <T> Iterable<T> insert(Iterable<T> entities, Duration ttl);
+
+    /**
+     * Modifies an entity that already exists in the database.
+     *
+     * <p>For an update to be made, a matching entity with the same unique identifier
+     * must be present in the database. In databases that use an append model to write data or
+     * follow the BASE model, this method behaves the same as the {@link #insert} method.</p>
+     *
+     * <p>If the entity is versioned (for example, with {@code jakarta.persistence.Version} or by
```

Comment:
Remove reference to jakarta.persistence class. Does NoSQL have an equivalent?
```suggestion
     * <p>If the entity is versioned (for example, with an annotation or by
```